src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -1,42 +1,45 @@
 package ru.hogwarts.school.service;
 
 import org.springframework.stereotype.Service;
+import ru.hogwarts.school.exceptions.NotFoundObject;
 import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.repository.FacultyRepository;
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashMap;
+import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Service
 public class FacultyService {
+    private final FacultyRepository facultyRepository;
 
-    HashMap<Long, Faculty> faculties = new HashMap<>();
-
-    long lastId = 0;
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
 
-    public Faculty createFaculty(Faculty faculty){
-        faculty.setId(++lastId);
-        faculties.put(lastId,faculty);
+    public Faculty createFaculty(Faculty faculty) {
+        facultyRepository.save(faculty);
         return faculty;
```

Comment:
```suggestion
    public Faculty createFaculty(Faculty faculty) {
        return facultyRepository.save(faculty);
```
Нужно вернуть именно тот объект, что вернул метод save